source/jobs.c

```diff
@@ -745,3 +826,136 @@ JobsStatus_t Jobs_Update( char * buffer,
+                               const size_t thingNameLength )
+{
+    return isThingnameTopicMatch( topic, topicLength, "/jobs/start-next/accepted", strlen( "/jobs/start-next/accepted" ), thingName, thingNameLength );
+}
+
+bool Jobs_IsJobUpdateStatus( const char * topic,
+                             const size_t topicLength,
+                             const char * jobId,
+                             const size_t jobIdLength,
+                             const char * thingName,
+                             const size_t thingNameLength,
+                             JobUpdateStatus_t expectedStatus )
+{
+    /* Max suffix size = max topic size - "$aws/<thingname>" prefix */
+    char suffixBuffer[ TOPIC_BUFFER_SIZE - MAX_THING_NAME_LENGTH - 4U ] = { 0 };
+    char jobIdTerminated[ JOBS_JOBID_MAX_LENGTH + 1 ] = { 0 };
+    char updateStatusString[ UPDATE_JOB_STATUS_MAX_LENGTH + 1 ] = { 0 };
+
+    memcpy( &jobIdTerminated, jobId, jobIdLength );
+    memcpy( &updateStatusString, jobUpdateStatusString[ expectedStatus ], jobUpdateStatusStringLengths[ expectedStatus ] );
+
+    snprintf( suffixBuffer,
+              TOPIC_BUFFER_SIZE - MAX_THING_NAME_LENGTH - 4U,
+              "%s%s%s%s",
+              "/jobs/",
+              jobIdTerminated,
+              "/update/",
+              updateStatusString );
+
+    return isThingnameTopicMatch( topic, topicLength, suffixBuffer, strnlen( suffixBuffer, TOPIC_BUFFER_SIZE - MAX_THING_NAME_LENGTH - 4U ), thingName, thingNameLength );
+}
+
+size_t Jobs_GetJobId( const char * message,
+                      size_t messageLength,
+                      char ** jobId )
+{
+    size_t jobIdLength = 0U;
+    JSONStatus_t jsonResult = JSONNotFound;
+
+    jsonResult = JSON_Validate( message, messageLength );
+
+    if( jsonResult == JSONSuccess )
+    {
+        jsonResult = JSON_Search( ( char * ) message,
+                                  messageLength,
+                                  "execution.jobId",
+                                  sizeof( "execution.jobId" ) - 1,
+                                  jobId,
+                                  &jobIdLength );
+    }
+
+    return jobIdLength;
+}
+
+size_t Jobs_GetJobDocument( const char * message,
+                            size_t messageLength,
+                            char ** jobDoc )
+{
+    size_t jobDocLength = 0U;
+    JSONStatus_t jsonResult = JSONNotFound;
+
+    jsonResult = JSON_Validate( message, messageLength );
+
+    if( jsonResult == JSONSuccess )
+    {
+        jsonResult = JSON_Search( ( char * ) message,
+                                  messageLength,
+                                  "execution.jobDocument",
+                                  sizeof( "execution.jobDocument" ) - 1,
+                                  jobDoc,
+                                  &jobDocLength );
+    }
+
+    return jobDocLength;
+}
+
+size_t Jobs_GetStartNextPendingJobExecutionMsg( const char * clientToken,
```

Comment:
Is the matching API to this `Jobs_StartNext`? Because if so, we should place these APIs closer to each other in code and have more consistent naming between them.

Reply:
Updated.